src/openrct2/core/IStream.cpp

```diff
@@ -61,4 +62,22 @@ namespace OpenRCT2
         WriteString(str.c_str());
     }
 
+    ObjectEntryDescriptor IStream::ReadObjectEntryDescriptor()
```

Comment:
This is not how any other custom structure reads and writes to the IStream. Its polluting the IStream with details that are relevant to a very small subset of the program.